modules/imgcodecs/test/test_animation.cpp

```diff
@@ -543,6 +543,50 @@ TEST(Imgcodecs_APNG, imencode_rgba)
     EXPECT_EQ(read_frames.size(), s_animation.frames.size() - 2);
 }
 
+typedef testing::TestWithParam<string> Imgcodecs_ImageCollection;
+
+const string exts_multi[] = {
+#ifdef HAVE_AVIF
+    ".avif",
+#endif
+#ifdef HAVE_IMGCODEC_GIF
+    ".gif",
+#endif
+    ".png",
+#ifdef HAVE_TIFF
+    ".tiff",
+#endif
+#ifdef HAVE_WEBP
+    ".webp",
+#endif
+};
+
+TEST_P(Imgcodecs_ImageCollection, animations)
+{
+    Animation s_animation;
+    EXPECT_TRUE(fillFrames(s_animation, false));
+
+    string output = cv::tempfile(GetParam().c_str());
+    ASSERT_TRUE(imwritemulti(output, s_animation.frames));
+    vector<Mat> read_frames;
+    ASSERT_TRUE(imreadmulti(output, read_frames, IMREAD_UNCHANGED));
+
+    {
```

Comment:
i used this block to make `ImageCollection` a local variable. otherwise `EXPECT_EQ(0, remove(output.c_str()));` fails